drivers/ethernet/eth_nxp_enet_qos/eth_nxp_enet_qos_mac.c

```diff
@@ -0,0 +1,606 @@
+		/* Start of tx descriptors buffer */
+		ENET_QOS_REG_PREP(DMA_CH_DMA_CHX_TXDESC_LIST_ADDR, TDESLA,
+			ENET_QOS_ALIGN_ADDR_SHIFT((uint32_t)tx->descriptors));
+	base->DMA_CH[0].DMA_CHX_TXDESC_TAIL_PTR =
+		/* Do not move the tail pointer past the start until send is requested */
+		ENET_QOS_REG_PREP(DMA_CH_DMA_CHX_TXDESC_TAIL_PTR, TDTP,
+			ENET_QOS_ALIGN_ADDR_SHIFT((uint32_t)tx->descriptors));
+	base->DMA_CH[0].DMA_CHX_TXDESC_RING_LENGTH =
+		ENET_QOS_REG_PREP(DMA_CH_DMA_CHX_TXDESC_RING_LENGTH, TDRL, NUM_TX_BUFDESC);
+}
+
+static inline int enet_qos_rx_desc_init(enet_qos_t *base, struct nxp_enet_qos_rx_data *rx)
+{
+	struct net_buf *buf;
+
+	memset((void *)rx->descriptors, 0, sizeof(union nxp_enet_qos_rx_desc) * NUM_RX_BUFDESC);
+
+	/* Here we reserve an RX buffer for each of the DMA descriptors. */
+	for (int i = 0; i < NUM_RX_BUFDESC; i++) {
+		buf = net_pkt_get_reserve_rx_data(CONFIG_NET_BUF_DATA_SIZE, K_NO_WAIT);
+		if (buf == NULL) {
+			LOG_ERR("Missing a buf");
+			return -ENOMEM;
+		}
+		rx->reserved_bufs[i] = buf;
+		rx->descriptors[i].read.buf1_addr = (uint32_t)buf->data;
+		rx->descriptors[i].read.control |= rx_desc_refresh_flags;
+	}
+
+	/* Set up RX descriptors on channel 0 */
+	base->DMA_CH[0].DMA_CHX_RXDESC_LIST_ADDR =
+		/* Start of tx descriptors buffer */
+		ENET_QOS_REG_PREP(DMA_CH_DMA_CHX_RXDESC_LIST_ADDR, RDESLA,
+			ENET_QOS_ALIGN_ADDR_SHIFT((uint32_t)&rx->descriptors[0]));
+	base->DMA_CH[0].DMA_CHX_RXDESC_TAIL_PTR =
+		/* When the DMA reaches the tail pointer, it suspends. Set to last descriptor */
+		ENET_QOS_REG_PREP(DMA_CH_DMA_CHX_RXDESC_TAIL_PTR, RDTP,
+			ENET_QOS_ALIGN_ADDR_SHIFT((uint32_t)&rx->descriptors[NUM_RX_BUFDESC]));
+	base->DMA_CH[0].DMA_CHX_RX_CONTROL2 =
+		/* Ring length == Buffer size. Register is this value minus one. */
+		ENET_QOS_REG_PREP(DMA_CH_DMA_CHX_RX_CONTROL2, RDRL, NUM_RX_BUFDESC - 1);
+	base->DMA_CH[0].DMA_CHX_RX_CTRL |=
+		/* Set DMA receive buffer size. The low 2 bits are not entered to this field. */
+		ENET_QOS_REG_PREP(DMA_CH_DMA_CHX_RX_CTRL, RBSZ_13_Y, NET_ETH_MAX_FRAME_SIZE >> 2);
+
+	return 0;
+}
+
+static int eth_nxp_enet_qos_mac_init(const struct device *dev)
+{
+	const struct nxp_enet_qos_mac_config *config = dev->config;
+	struct nxp_enet_qos_mac_data *data = dev->data;
+	struct nxp_enet_qos_config *module_cfg = ENET_QOS_MODULE_CFG(config->enet_dev);
+	enet_qos_t *base = module_cfg->base;
+	uint32_t clk_rate;
+	int ret;
+
+	/* Used to configure timings of the MAC */
+	ret = clock_control_get_rate(module_cfg->clock_dev, module_cfg->clock_subsys, &clk_rate);
+	if (ret) {
+		return ret;
+	}
+
+	/* For reporting the status of the link connection */
+	ret = phy_link_callback_set(config->phy_dev, eth_nxp_enet_qos_phy_cb, (void *)dev);
+	if (ret) {
+		return ret;
+	}
+
+	/* Random mac therefore overrides local mac that may have been initialized */
+	if (config->random_mac) {
+		gen_random_mac(data->mac_addr.addr,
+			NXP_OUI_BYTE_0, NXP_OUI_BYTE_1, NXP_OUI_BYTE_2);
```

Comment:
indentation

Reply:
can you clarify what the indentation should be?

Reply:
@decsny, the second line should just tab to the beginning of the first line then use spaces to align the rest.